pkg/components/ebpf/common/tcp_detect_transform.go

```diff
@@ -148,6 +120,118 @@ func ReadTCPRequestIntoSpan(parseCtx *EBPFParseContext, cfg *config.EBPFTracer,
 	return request.Span{}, true, nil // ignore if we couldn't parse it
 }
 
+func fixupBuffers(parseCtx *EBPFParseContext, event *TCPRequestInfo) (req []byte, resp []byte) {
+	l := int(event.Len)
+	if l < 0 || len(event.Buf) < l {
+		l = len(event.Buf)
+	}
+	req = event.Buf[:l]
+
+	l = int(event.RespLen)
+	if l < 0 || len(event.Rbuf) < l {
+		l = len(event.Rbuf)
+	}
+	resp = event.Rbuf[:l]
+
+	if event.HasLargeBuffers == 1 {
+		if b, ok := getTCPLargeBuffer(parseCtx, event.Tp.TraceId, event.Tp.SpanId, 0); ok {
+			req = b
+		}
+		if b, ok := getTCPLargeBuffer(parseCtx, event.Tp.TraceId, event.Tp.SpanId, 1); ok {
+			resp = b
+		}
+	}
+
+	return
+}
+
+func handleMySQL(parseCtx *EBPFParseContext, event *TCPRequestInfo, requestBuffer, responseBuffer []byte) (request.Span, error) {
+	var span request.Span
+
+	if len(requestBuffer) < sqlprune.MySQLHdrSize+1 {
+		slog.Warn("MySQL request too short")
+		return span, errFallback
+	}
+
+	stmt := string(requestBuffer[sqlprune.MySQLHdrSize+1:])
+	sqlCommand := sqlprune.SQLParseCommandID(request.DBMySQL, requestBuffer)
+	if sqlCommand == "" {
+		slog.Warn("MySQL command ID unhandled", "commandID", requestBuffer[sqlprune.MySQLHdrSize])
+		return span, errIgnore
+	}
+
+	sqlError := sqlprune.SQLParseError(responseBuffer)
+
+	switch sqlCommand {
+	case "STMT_PREPARE":
+		if sqlError != nil {
+			slog.Debug("MySQL PREPARE command errored, ignoring", "error", sqlError)
+			return span, errIgnore
+		}
+
+		// On the PREPARE command, the statement ID is the first 4 bytes after the header and command ID
+		// in the response buffer.
+		stmtID := sqlprune.SQLParseStatementID(request.DBMySQL, responseBuffer)
+		if stmtID == 0 {
+			slog.Warn("MySQL PREPARE command with invalid statement ID")
+			return span, errFallback
+		}
+
+		stmt = string(requestBuffer[sqlprune.MySQLHdrSize+1:])
+
+		parseCtx.mysqlPreparedStatements.Add(mysqlPreparedStatementsKey{
+			connInfo: event.ConnInfo,
+			stmtID:   stmtID,
+		}, stmt)
+
+		return span, errIgnore
+	case "STMT_EXECUTE":
+		if sqlError != nil {
```

Comment:
if execute fails isn't this like if a query fails? and thats a usecase we dont want to ignore

Reply:
nice catch, copy paste gone wrong